app/src/androidTest/java/com/aws/amazonlocation/ui/main/SettingsFragmentAnalyticsTest.kt

```diff
@@ -0,0 +1,141 @@
+package com.aws.amazonlocation.ui.main
+
+import android.app.ActivityManager
+import android.content.Context
+import androidx.constraintlayout.widget.ConstraintLayout
+import androidx.test.core.app.ApplicationProvider
+import androidx.test.espresso.Espresso.onView
+import androidx.test.espresso.action.ViewActions
+import androidx.test.espresso.action.ViewActions.click
+import androidx.test.espresso.assertion.ViewAssertions
+import androidx.test.espresso.matcher.ViewMatchers.* // ktlint-disable no-wildcard-imports
+import androidx.test.platform.app.InstrumentationRegistry.getInstrumentation
+import androidx.test.uiautomator.By
+import androidx.test.uiautomator.UiDevice
+import androidx.test.uiautomator.Until
+import com.aws.amazonlocation.* // ktlint-disable no-wildcard-imports
+import com.aws.amazonlocation.di.AppModule
+import com.aws.amazonlocation.utils.EventType
+import com.aws.amazonlocation.utils.KEY_MAP_NAME
+import com.aws.amazonlocation.utils.KEY_MAP_STYLE_NAME
+import com.aws.amazonlocation.utils.PreferenceManager
+import com.google.android.material.bottomsheet.BottomSheetBehavior
+import dagger.hilt.android.testing.HiltAndroidTest
+import dagger.hilt.android.testing.UninstallModules
+import org.hamcrest.core.AllOf.allOf
+import org.junit.*
+
+// ktlint-disable no-wildcard-imports
+
+@UninstallModules(AppModule::class)
+@HiltAndroidTest
+class SettingsFragmentAnalyticsTest : BaseTestMainActivity() {
+
+    private val uiDevice = UiDevice.getInstance(getInstrumentation())
+
+    private lateinit var preferenceManager: PreferenceManager
+
+    @Throws(java.lang.Exception::class)
+    override fun before() {
+        preferenceManager = PreferenceManager(ApplicationProvider.getApplicationContext())
+        preferenceManager.removeValue(KEY_MAP_NAME)
+        preferenceManager.removeValue(KEY_MAP_STYLE_NAME)
+
+        super.before()
+    }
+
+    @Test
+    fun checkContent() {
```

Comment:
< 2 mins, This test will check the success record of the event, it's based on UI update, On the successful record it will show the snack bar (only during the testing process) and it will verify with snack bar details.